src/terraform-resource/out/out_lifecycle_test.go

```diff
@@ -178,4 +181,248 @@ var _ = Describe("Out Lifecycle", func() {
+					Source: "fixtures/aws/",
+					Vars: map[string]interface{}{
+						"access_key":     accessKey,
+						"secret_key":     secretKey,
+						"bucket":         bucket,
+						"object_key":     s3ObjectPath,
+						"object_content": "terraform-is-neat",
+						"region":         region,
+					},
+				},
+			},
+		}
+
+		applyRequest := models.OutRequest{
+			Source: models.Source{
+				Storage: storage.Model{
+					Bucket:          bucket,
+					BucketPath:      bucketPath,
+					AccessKeyID:     accessKey,
+					SecretAccessKey: secretKey,
+					RegionName:      region,
+				},
+			},
+			Params: models.OutParams{
+				PlanRun: true,
+				EnvName: envName,
+				Terraform: models.Terraform{
+					Source: "fixtures/aws/",
+					Vars: map[string]interface{}{
+						"access_key":     accessKey,
+						"secret_key":     secretKey,
+						"bucket":         bucket,
+						"object_key":     s3ObjectPath,
+						"object_content": "terraform-is-neat",
+						"region":         region,
+					},
+				},
+			},
+		}
+
+		By("ensuring plan file does not already exist")
+
+		awsVerifier.ExpectS3FileToNotExist(
+			planOutRequest.Source.Storage.Bucket,
+			planFilePath,
+		)
+
+		By("running 'out' to create the plan file")
+
+		planrunner := out.Runner{
+			SourceDir: workingDir,
+			LogWriter: GinkgoWriter,
+		}
+		planOutput, err := planrunner.Run(planOutRequest)
+		Expect(err).ToNot(HaveOccurred())
+
+		By("ensuring that plan file exists with valid version (LastModified)")
+
+		awsVerifier.ExpectS3FileToExist(
+			planOutRequest.Source.Storage.Bucket,
+			planFilePath,
+		)
+
+		_, err = time.Parse(storage.TimeFormat, planOutput.Version.LastModified)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(planOutput.Version.EnvName).To(Equal(planOutRequest.Params.EnvName))
+
+		time.Sleep(1 * time.Second) // ensure LastModified changes
+
+		By("ensuring state file does not already exist")
+
+		awsVerifier.ExpectS3FileToNotExist(
+			applyRequest.Source.Storage.Bucket,
+			stateFilePath,
+		)
+
+		By("applying the plan")
+
+		applyrunner := out.Runner{
+			SourceDir: workingDir,
+			LogWriter: GinkgoWriter,
+		}
+		createOutput, err := applyrunner.Run(applyRequest)
+		Expect(err).ToNot(HaveOccurred())
+
+		Expect(createOutput.Metadata).ToNot(BeEmpty())
+		fields := map[string]interface{}{}
+		for _, field := range createOutput.Metadata {
+			fields[field.Name] = field.Value
+		}
+		Expect(fields["env_name"]).To(Equal(envName))
+		expectedMD5 := fmt.Sprintf("%x", md5.Sum([]byte("terraform-is-neat")))
+		Expect(fields["content_md5"]).To(Equal(expectedMD5))
+
+		awsVerifier.ExpectS3FileToExist(
+			applyRequest.Source.Storage.Bucket,
+			s3ObjectPath,
+		)
+
+		By("ensuring that plan file no longer exists after the apply")
+
+		awsVerifier.ExpectS3FileToNotExist(
+			applyRequest.Source.Storage.Bucket,
+			planFilePath,
+		)
+		Expect(err).ToNot(HaveOccurred())
+
+	})
+
+	It("plan should be deleted on destroy", func() {
+		applyNoPlanRequest := models.OutRequest{
```

Comment:
Probably don't need this first apply. Running `plan` then `destroy` and ensuring the statefile is gone should be fine.